cpp/include/cudf/io/json.hpp

```diff
@@ -354,7 +354,24 @@ class json_reader_options {
    *
    * @param delimiter Delimiter separating records in JSON lines
    */
-  void set_delimiter(char delimiter) { _delimiter = delimiter; }
+  void set_delimiter(char delimiter)
+  {
+    switch (delimiter) {
+      case '{': [[fallthrough]];
+      case '[': [[fallthrough]];
+      case '}': [[fallthrough]];
+      case ']': [[fallthrough]];
+      case ',': [[fallthrough]];
+      case ':': [[fallthrough]];
+      case '"': [[fallthrough]];
+      case '\'': [[fallthrough]];
+      case '\\': [[fallthrough]];
+      case ' ': [[fallthrough]];
+      case '\t': [[fallthrough]];
```

Comment:
As I learned recently, this is only recommended when the case in not empty
```suggestion
      case '{':
      case '[':
      case '}':
      case ']':
      case ',':
      case ':':
      case '"':
      case '\'':
      case '\\':
      case ' ':
      case '\t':
```

Reply:
Didn't know this, thanks!